pkg/kubeapiserver/options/authorization.go

```diff
@@ -60,6 +66,12 @@ func (s *BuiltInAuthorizationOptions) AddFlags(fs *pflag.FlagSet) {
 		"File with webhook configuration in kubeconfig format, used with --authorization-mode=Webhook. "+
 		"The API server will query the remote service to determine access on the API server's secure port.")
 
+	fs.BoolVar(&s.WebhookUseServiceResolution, "authorization-webhook-use-service-resolution", s.WebhookUseServiceResolution,
+		"If set, the authorizer webhook code will resolve URLs pointing to hosts of the form "+
+			"https://servicename.namespace.svc:port/somewhere to appropriate endpoint IP addresses. "+
+			"You may want to use this feature if your webhook authorizer is hosted in the cluster itself, "+
+			"and the apiserver can not access kube-dns.")
```

Comment:
Phrase this more generically... `kube-dns` is only one possible in-cluster DNS implementation